src/xregexp.js

```diff
@@ -238,6 +238,14 @@ function getContextualTokenSeparator(match, scope, flags) {
         // No need to separate tokens if at the beginning or end of a group
         match.input[match.index - 1] === '(' ||
         match.input[match.index + match[0].length] === ')' ||
+        // No need to separate tokens if at the beginning of a non-capturing group
+        match.input.slice(match.index - 3, 3) === '(?:' ||
+        // No need to separate tokens if before or after a `|`
+        match.input[match.index - 1] === '|' ||
+        match.input[match.index + match[0].length] === '|' ||
+        // No need to separate tokens if at the beginning or end of the pattern
+        match.input[match.index - 1] === undefined ||
+        match.input[match.index + match[0].length] === undefined ||
```

Comment:
Ditto. Should probably be comparing to the length of `match.input` rather than referencing a specific character in the string.

Reply:
Also fixed in https://github.com/slevithan/xregexp/commit/ee766c211250cc4c90d73123ff1d6095007f0c96